src/App.tsx

```diff
@@ -1,14 +1,95 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+export type FilterProperties = {
+  filterType: 'all' | 'active' | 'completed';
+  filterValue: string;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [userTodo, setUserTodo] = useState<Todo | null>(null);
+  const [user, setUser] = useState<User | null>(null);
```

Comment:
You could keep the logic and state of user inside Modal and load i using `useEffect` there.